src/main/java/com/money_account_service/services/AccountService.java

```diff
@@ -2,30 +2,56 @@
 
 
 import com.money_account_service.dtos.request.CreateAccountRequestDto;
+import com.money_account_service.dtos.response.AccountDetailsResponseDto;
 import com.money_account_service.dtos.response.CreateAccountResponseDto;
+import com.money_account_service.entities.AccountEntity;
+import com.money_account_service.mappers.ResponseMapper;
 import com.money_account_service.models.UserModel;
 import com.money_account_service.repositories.AccountRepository;
-import com.money_account_service.utility.UserServiceClient;
+import com.money_account_service.utility.DateTimeProvider;
 import lombok.AllArgsConstructor;
+import org.iban4j.CountryCode;
+import org.iban4j.Iban;
+import org.springframework.http.HttpStatus;
+import org.springframework.web.server.ResponseStatusException;
+
+import java.util.Optional;
 
 @AllArgsConstructor
 public class AccountService {
 
     private AccountRepository accountRepository;
 
+    private DateTimeProvider dateTimeProvider;
+
     public CreateAccountResponseDto createAccount(CreateAccountRequestDto createAccountRequestDto, UserModel userModel) {
```

Comment:
We should start adding transactionality to methods, which modify the DB state. https://www.baeldung.com/transaction-configuration-with-jpa-and-spring